src/PandoraMonitoring.cc

```diff
@@ -783,15 +838,43 @@ TEveElement *PandoraMonitoring::VisualizeVertices(const pandora::VertexList *con
     TEvePointSet *pTEvePointSet = new TEvePointSet(name.c_str(), 1);
     pTEvePointSet->SetOwnIds(kTRUE);
 
-    for (VertexList::const_iterator iter = pVertexList->begin(), iterEnd = pVertexList->end(); iter != iterEnd; ++iter)
+    bool isU(VERTEX_U == pVertexList->front()->GetVertexType());
+    bool isV(VERTEX_V == pVertexList->front()->GetVertexType());
+    bool isW(VERTEX_W == pVertexList->front()->GetVertexType());
+
+    try
     {
-        const Vertex *pVertex = *iter;
-        const CartesianVector &vertexPosition(pVertex->GetPosition());
+        for (VertexList::const_iterator iter = pVertexList->begin(), iterEnd = pVertexList->end(); iter != iterEnd; ++iter)
+        {
+            const Vertex *pVertex = *iter;
+            const CartesianVector &vertexPosition(pVertex->GetPosition());
+            VertexType vertexType(pVertex->GetVertexType());
 
-        std::stringstream sstr;
-        sstr << starter << "Vertex\n" << vertexPosition;
+            if ((isU && VERTEX_U != vertexType) || (isV && VERTEX_V != vertexType) || (isW && VERTEX_W != vertexType))
+                throw StatusCodeException(STATUS_CODE_FAILURE);
 
-        (void) AddMarkerToVisualization(&vertexPosition, sstr.str(), pTEvePointSet, color, 1);
+            std::stringstream sstr;
+            sstr << starter << "Vertex\n" << vertexPosition;
+            const std::string markerName(sstr.str());
+            const std::string markerTitle(markerName.empty() ? "Marker" : markerName);
+            const int markerSize(1);
+            const int markerStyle(20);
+
+            TEvePointSet *pTEvePointSetMarker = new TEvePointSet(markerTitle.c_str(), 1);
+            pTEvePointSetMarker->SetOwnIds(kTRUE);
+            pTEvePointSetMarker->SetPoint(0, vertexPosition.GetX() * m_scalingFactor, vertexPosition.GetY() * m_scalingFactor, vertexPosition.GetZ() * m_scalingFactor);
+
+            const Color chosenColor((color < AUTO) ? color : ORANGE);
+            pTEvePointSetMarker->SetMarkerColor(GetROOTColor(chosenColor));
+            pTEvePointSetMarker->SetMarkerSize(markerSize);
+            pTEvePointSetMarker->SetMarkerStyle(markerStyle);
+
+            pTEvePointSet->AddElement(pTEvePointSetMarker);
+        }
+    }
+    catch (StatusCodeException &)
```

Comment:
We should get into the habit of catching exceptions by const reference e.g. 'catch (const StatusCodeException &)'